lib/puppetfile-resolver/resolution_provider.rb

```diff
@@ -137,23 +141,31 @@ def find_all_module_specifications(dependency)
       unless mod.nil?
         case mod.module_type
         when Puppetfile::FORGE_MODULE
-          @module_info[dependency.name] = safe_spec_search(dependency) { SpecSearchers::Forge.find_all(dependency, @cache, @resolver_ui) }
+          @module_info[dependency.name] = safe_spec_search(dependency) do
+            SpecSearchers::Forge.find_all(dependency, @cache, @forge_api_url, @forge_proxy, @resolver_ui)
+          end
```

Comment:
Originally thought of wrapping these calls in a `with_proxy` block, but that made adding the proxy to any error messages a little weirder. Opted to instead pass the proxy along to the spec searcher for now.